GitHubContents.ps1

```diff
@@ -143,3 +153,266 @@ function Get-GitHubContent
+    {
+        if (![System.String]::IsNullOrEmpty($CommitterName) -and
+            ![System.String]::IsNullOrEmpty($CommitterEmail))
+        {
+            $hashBody['committer'] = @{
+                name = $CommitterName
+                email = $CommitterEmail
+            }
+        }
+        else
+        {
+            throw 'Both CommiterName and CommitterEmail need to be specified.'
+        }
+    }
+
+    if ($PSBoundParameters.ContainsKey('AuthorName') -or
+        $PSBoundParameters.ContainsKey('AuthorEmail'))
+    {
+        if (![System.String]::IsNullOrEmpty($CommitterName) -and
+            ![System.String]::IsNullOrEmpty($CommitterEmail))
+        {
+            $hashBody['author'] = @{
+                name = $AuthorName
+                email = $AuthorEmail
+            }
+        }
+        else
+        {
+            throw 'Both AuthorName and AuthorEmail need to be specified.'
+        }
+    }
+
+    if ($PSCmdlet.ShouldProcess("$BranchName branch of $RepositoryName",
+        "Set GitHub Contents on $Path"))
+    {
+        Write-InvocationLog
+
+        $params = @{
+            UriFragment = $uriFragment
+            Description = "Writing content for $Path in the $BranchName branch of $RepositoryName"
+            Body = (ConvertTo-Json -InputObject $hashBody)
+            Method = 'Put'
+            AccessToken = $AccessToken
+            TelemetryEventName = $MyInvocation.MyCommand.Name
+            TelemetryProperties = $telemetryProperties
+            NoStatus = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus `
+                -ConfigValueName DefaultNoStatus)
+        }
+
+        try
+        {
+            return Invoke-GHRestMethod @params
+        }
+        catch
+        {
+            $overwriteShaRequired = $false
+
+            # Temporary code to handle current differences in exception object between PS5 and PS7
+            if ($PSVersionTable.PSedition -eq 'Core')
+            {
+                $errorMessage = ($_.ErrorDetails.Message | ConvertFrom-Json).message -replace '\n',' ' -replace '\"','"'
+                if (($_.Exception -is [Microsoft.PowerShell.Commands.HttpResponseException]) -and
+                    ($errorMessage -eq 'Invalid request.  "sha" wasn''t supplied.'))
+                {
+                    $overwriteShaRequired = $true
+                }
+                else
+                {
+                    throw $_
+                }
+            }
+            else
+            {
+                $errorMessage = $_.Exception.Message  -replace '\n',' ' -replace '\"','"'
+                if ($errorMessage -like '*Invalid request.  "sha" wasn''t supplied.*')
+                {
+                    $overwriteShaRequired = $true
+                }
+                else
+                {
+                    throw $_
+                }
+            }
+
+            if ($overwriteShaRequired)
+            {
+                # Get SHA from current file
```

Comment:
This scenario warrants a verbose log entry to explain what's going on...something like:
```powershell
Write-Log -Level Verbose -Message 'Replacing the content of an existing file requires the current SHA of that file.  Retrieving the SHA now.'
```